### PR TITLE
Features/router module

### PR DESCRIPTION
**Pull request by curtwl**
# Description

Replaced the existing AppModule structure with a RouterModule structure that places all modules with a common "teams" path into the TeamsModule.  Currently this includes ResourcesModule, TechsModule, FeaturesModule, and IdeationsModule.

## Issue link

Fixes # [30](https://github.com/chingu-x/chingu-dashboard-be/issues/30)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested endpoints with Swagger.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by cherylli** [comment]
I thought we talked about in the meeting yesterday to have voyage instead of teams
or are we doing something like voyage/teams

File: src/app.module.ts [hunk 1]
```diff
@@ -1,31 +1,42 @@
...
+import { ResourcesModule } from "./resources/resources.module";
+import { TechsModule } from "./techs/techs.module";
+import { FeaturesModule } from "./features/features.module";
+import { IdeationsModule } from "./ideations/ideations.module";
 
 @Module({
     imports: [
+        RouterModule.register([
+            {
+                path: "voyage",
```
Comment (jenny-alexander) [inline src/app.module.ts:21]: Can you modify this to be voyages (plural) to follow the API naming standard?

**Review by jenny-alexander — DISMISSED** [review]
LGTM

**Review by cherylli — COMMENTED** [review]
looks good but I have a suggestion - rename these "endpoint group headers" to "voyage - Ideation" or "Voyage/Ideation" etc, so they are all grouped together on swagger (atm they are sorted alphabetically)
image

**Comment by cherylli** [comment]
should sprints be under voyage too?

**Comment by curtwl** [comment]
> should sprints be under voyage too?

I was conflicted about that since the sprints endpoints are clearly related to voyages, but don't have the common "voyages" path. Should I go ahead and group sprints under voyages too?

**Comment by cherylli** [comment]
> > should sprints be under voyage too?
> 
> I was conflicted about that since the sprints endpoints are clearly related to voyages, but don't have the common "voyages" path. Should I go ahead and group sprints under voyages too?

I think so, I think the point of adding router module is to group resources together

This shows what goes under "voyage" (from the figma design)
image

**Comment by curtwl** [comment]
> > > should sprints be under voyage too?
> > 
> > 
> > I was conflicted about that since the sprints endpoints are clearly related to voyages, but don't have the common "voyages" path. Should I go ahead and group sprints under voyages too?
> 
> I think so, I think the point of adding router module is to group resources together
> 
> This shows what goes under "voyage" (from the figma design) image

Done. If we wanted to group sprints with the other voyage modules in the RouterModule, we'd have to add the "voyages" path. But they are grouped together in the Swagger docs at least.

**Comment by cherylli** [comment]
> > > > should sprints be under voyage too?
> > > 
> > > 
> > > I was conflicted about that since the sprints endpoints are clearly related to voyages, but don't have the common "voyages" path. Should I go ahead and group sprints under voyages too?
> > 
> > 
> > I think so, I think the point of adding router module is to group resources together
> > This shows what goes under "voyage" (from the figma design) image
> 
> Done. If we wanted to group sprints with the other voyage modules in the RouterModule, we'd have to add the "voyages" path. But they are grouped together in the Swagger docs at least.

yes I think we should do that since it's a part of voyages. Unless there's a reason why we should not do that?

**Comment by curtwl** [comment]
> > > > > should sprints be under voyage too?
> > > > 
> > > > 
> > > > I was conflicted about that since the sprints endpoints are clearly related to voyages, but don't have the common "voyages" path. Should I go ahead and group sprints under voyages too?
> > > 
> > > 
> > > I think so, I think the point of adding router module is to group resources together
> > > This shows what goes under "voyage" (from the figma design) image
> > 
> > 
> > Done. If we wanted to group sprints with the other voyage modules in the RouterModule, we'd have to add the "voyages" path. But they are grouped together in the Swagger docs at least.
> 
> yes I think we should do that since it's a part of voyages. Unless there's a reason why we should not do that?

Looks like I misunderstood you when you said "should sprints be under voyage too?" because of the context, I thought you just meant the Swagger docs. I agree, I just assumed that maybe frontend wanted the path to be /api/v1/sprints and didn't ask, oops. Will change that to /api/v1/voyages/sprints with the RouterModule then.

**Review by jenny-alexander — APPROVED** [review]
Changes to put sprints under voyages look good.

**Review by cherylli — APPROVED** [review]
thanks!

**Pull request merged**